src/main/java/org/gridsuite/shortcircuit/server/service/ShortCircuitService.java

```diff
@@ -101,6 +97,15 @@ private static FeederResult fromEntity(FeederResultEmbeddable feederResultEmbedd
         return new FeederResult(feederResultEmbeddable.getConnectableId(), feederResultEmbeddable.getCurrent());
     }
 
+    private static ShortCircuitAnalysisResultEntity sortByElementId (ShortCircuitAnalysisResultEntity result) {
+        List<FaultResultEntity> sortedFaultResults = result.getFaultResults().stream()
+                .sorted(Comparator.comparing(fr -> fr.getFault().getElementId()))
+                .toList();
+        Set<FaultResultEntity> sortedFaultResultsSet = new LinkedHashSet<>(sortedFaultResults);
+        result.setFaultResults(sortedFaultResultsSet);
+        return result;
+    }
```

Comment:
How about :
```suggestion
        private static ShortCircuitAnalysisResultEntity sortByElementId (ShortCircuitAnalysisResultEntity result) {
        result.setFaultResults(result.getFaultResults().stream()
                .sorted(Comparator.comparing(fr -> fr.getFault().getElementId()))
                .collect(Collectors.toCollection(LinkedHashSet::new)));
        return result;
    }
```
It would be more concise